dlrover/trainer/torch/flash_checkpoint/engine.py

```diff
@@ -184,13 +188,20 @@ def __init__(
             )
         else:
             self._event_queue = None  # type: ignore
+        self._update_saver_config()
+
         # lock for shared memory
         local_shard_num = self.get_local_shard_num()
         self.local_shard_id = self._local_rank % local_shard_num
         lock_name = CheckpointSharedObjPrefix.SHM_LOCK_NAME + str(
             self.local_shard_id
         )
         self._shm_lock = SharedLock(name=lock_name, create=False)
+
+        # need to wait until socket server created(by the saver)
```

Comment:
# wait until socket server is created by the saver

Reply:
done